### ISSUE TITLE
Too large individual GC pause (~ 4 minutes) while total memory isn't large

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/runtime#97018]
### Description

**Note:** This may/may not be a runtime/GC bug.

We have an issue that an app gets extremely slow with time. It's not reproducible locally, only in CI. And specifically, only when running macOS-12.

After adding some logs (https://github.com/unoplatform/uno/commit/a60c3f15acfb4f3b88946b4ea43c1ad58aa8a299), we see extremely large individual GC pauses.

![image](https://github.com/dotnet/runtime/assets/31348972/4ddba5ef-e194-4421-92df-94d6a28c2041)

Given that the total memory isn't very high, there shouldn't be much of memory load to have such large individual GC pauses. I also think the issue wouldn't be Gen2 fragmentation (we have a code path that allocates on LOH heavily but commenting that out to avoid Gen2 fragmentation doesn't help that much).

In short, I find 4 minutes pause is extremely a large number that no way should happen especially that total memory is only 300 to 400MB.

Would that, somehow, be a GC bug? A bug on our side? How to collect more info?

**Comment by janvorli** [comment]
@Youssef1313 what is the runtime version you were using?

**Comment by Youssef1313** [comment]
.NET 7

**Comment by Youssef1313** [comment]
Reducing some allocations helped reduce pause times, but the app still froze after some time. CI is moved to Windows for now which seems to work fine.
